canonical-web-and-design/maas-ui
# Domain details table

## Done

- Added links to nodes from domain details table (when available)
- Added sorting to domain resources table
- Added actions menu to domain resources table (with actions disabled until implemented)

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- `yarn start`
- go to http://0.0.0.0:8400/MAAS/r/domains
- click on different domains
- make sure resources table on domain details page is sortable and has action menu
- go to http://0.0.0.0:8400/MAAS/r/domain/0
- resources in the table should be linked to nodes
- actions menu should be disabled (as these are generated rows)

## Fixes

Fixes: canonical-web-and-design/app-squad#42

<img width="1443" alt="Screenshot 2021-06-15 at 16 16 57" src="https://user-images.githubusercontent.com/83575/122069229-4ef08800-cdf5-11eb-9560-4d469f0bdf4d.png">
